src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -568,17 +568,32 @@ private static void CalculateExcludes(IncludeExcludeFiles files, out string full
         public static void AddDependencyGroups(
             IEnumerable<LibraryDependency> dependencies,
             NuGetFramework framework,
-            PackageBuilder builder)
+            PackageBuilder builder,
+            bool packPrivateAssetsFlow)
         {
             ISet<PackageDependency> packageDependencies = new HashSet<PackageDependency>();
 
             foreach (LibraryDependency dependency in dependencies)
             {
-                LibraryIncludeFlags effectiveInclude = dependency.IncludeType & ~dependency.SuppressParent;
+                LibraryIncludeFlags effectiveInclude = default;
+
+                if (packPrivateAssetsFlow)
+                {
+                    effectiveInclude = dependency.IncludeType | (~dependency.SuppressParent & LibraryIncludeFlags.All);
```

Comment:
Reasoning for `& LibraryIncludeFlags.All` to limit `~dependency.SuppressParent` to byte size. Otherwise `~none` would result in `65535` not `127`.

Reply:
Should this be just
`effectiveInclude = ~dependency.SuppressParent & LibraryIncludeFlags.All;`